src/nc_server.c

```diff
@@ -290,12 +290,13 @@ server_failure(struct context *ctx, struct server *server)
 
     server->failure_count = 0;
 
+    add_failed_server(ctx, server);
```

Comment:
LGTM - I'd independently run into the same bug and had done something similar as a part of 82d4bba1d64a71f533e4b5886d3a687e459451ba in a fork before I'd seen this when implementing support for recursive failover